imapclient/client_test.go

```diff
@@ -62,6 +63,27 @@ func newClientServerPair(t *testing.T, initialState imap.ConnState) (*imapclient
 	return client, server
 }
 
+// bad server: connected but without greeting
+func newClientWithBadServer(t *testing.T) (*imapclient.Client, io.Closer) {
```

Comment:
Since this is only used by a single example, and unlikely to be used by any other, maybe we should fold this function's logic into `TestWaitGreetingError`?